DomainServices/Services/CustomerServices.cs

```diff
@@ -1,74 +1,94 @@
 ﻿using DomainModels;
 using DomainServices.Interfaces;
+using EntityFrameworkCore.UnitOfWork.Interfaces;
 using Infrastructure.Data.Context;
-using Microsoft.EntityFrameworkCore;
 using System;
 using System.Collections.Generic;
-using System.Linq;
+using System.Linq.Expressions;
 
 namespace DomainServices
 {
     public class CustomerServices : ICustomerServices
     {
-        private readonly DatabaseDbContext _dbContext;
+        private readonly IUnitOfWork _unitOfWork;
 
-        public CustomerServices(DatabaseDbContext dbContext)
+        private readonly IRepositoryFactory _repositoryFactory;
+
+        public CustomerServices(IUnitOfWork<DatabaseContext> unitOfWork, IRepositoryFactory<DatabaseContext> repositoryFactory)
         {
-            _dbContext = dbContext ?? throw new ArgumentNullException(nameof(dbContext));
+            _unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
+            _repositoryFactory = repositoryFactory ?? throw new ArgumentNullException(nameof(repositoryFactory));
         }
 
-        public IEnumerable<Customer> GetAll(Predicate<Customer> predicate = null)
+        public IEnumerable<Customer> GetAll(Expression<Func<Customer, bool>> predicates)
         {
-            if (predicate is null) return _dbContext.Customers;
-            return _dbContext.Customers.AsEnumerable();
+            var repository = _repositoryFactory.Repository<Customer>();
+
+            var query = repository.MultipleResultQuery();
+            query.AndFilter(predicates);
+
+            return repository.Search(query);
         }
 
-        public Customer GetBy(Func<Customer, bool> predicate)
+        public Customer GetBy(Expression<Func<Customer, bool>> predicates)
```

Comment:
Como é apenas 1, colocamos no singular
```suggestion
        public Customer GetBy(Expression<Func<Customer, bool>> predicate)
```